async_cache.go

```diff
@@ -88,8 +126,20 @@ func (ks *keyStatus) deleteExpiredKeys() {
 	}
 }
 
+func (ks *keyStatus) lock() {
```

Comment:
at Line 123: deleteExpiredKeys() => check for IN_PROCESS even after expiry, can be major issue " Query COntext Execution-> go routine Blocked"

Reply:
this will be rare case so we will log such cases where after expiry of key still process status is IN_PROCESS